lib/pmd.js

```diff
@@ -280,9 +283,11 @@ pmd.generateData = function(config){
         (folder.source.fileFilterRegexp instanceof RegExp && folder.source.fileFilterRegexp.test(files[i])) ||
         !folder.source.fileFilterRegexp instanceof RegExp){
 
+        let newextname = path.extname(config.folders[0].template);
```

Comment:
A few changes:

- [ ] Better name for variable. Perhaps `docExtName`
- [ ] Since we're removing the hard coding of `.md` (which I agree is a good thing) we should look at the current template extension. In some cases (edge cases) this may not be the same throughout. At this line of code we could reference `folder.template` rather than `config.folders[0].template` which is more accurate.